src/core/olaf.ts

```diff
@@ -0,0 +1,70 @@
+import * as TelegramBot from 'node-telegram-bot-api';
+import { Message } from 'node-telegram-bot-api';
+
+import preferencesDashboard from '../dashboard/preferences-dashboard';
+import MessageSender from './messageSender';
+import MessageHandler from './messageHandler';
+import MessageRouter from './messageRouter';
+import EndUseCaseResponse from '../classes/EndUseCaseResponse';
+import UseCase from '../interfaces/useCase';
+
+export default class Olaf {
+  private telegramBot;
+
+  private dashboard;
+
+  private messageHandler;
+
+  private messageRouter;
+
+  private messageSender;
+
+  private activeUseCase: UseCase;
+
+  constructor() {
+    this.telegramBot = new TelegramBot(process.env.BOT_TOKEN, { polling: true });
+    this.dashboard = preferencesDashboard;
+    this.messageHandler = new MessageHandler(this.telegramBot);
+    this.messageRouter = new MessageRouter();
+    this.messageSender = new MessageSender(this.telegramBot);
+    this.activeUseCase = null;
+
+    // TODO register all use cases here
+    // this.messageRouter.registerUseCase(new XUseCase())
+  }
```

Comment:
A constructor should be used to initialise a class not a whole application

Reply:
What if the class is the whole application?
Hier werden die ganzen Dependencies injiziert, dann muss man nicht mit `getTelegramBotInstance` und so arbeiten, aber sonst wird außer der Registrierung der Use Cases nichts gemacht. Und das hat imho alles was im Konstruktor der Application zu suchen.